lib/value/src/kind/merge.rs

```diff
@@ -1,11 +1,39 @@
 //! All types related to merging one [`Kind`] into another.
 
+use crate::kind::nest::CoalescedPath;
 use crate::kind::Field;
+use lookup::Lookup;
 use std::{collections::BTreeMap, ops::BitOr};
 
+use super::nest;
 use super::{Collection, Kind};
 
 impl Kind {
+    /// Returns a new kind with the type at the specific path set
+    ///
+    /// # Errors
+    /// Because `nest_at_path` doesn't support coalescing or negative indices, an error will
+    /// be returned if a path is used that contains either of those. In the future those should
+    /// be supported so this functon can just return `Kind` instead of `Result`
```

Comment:
```suggestion
    /// Returns a new kind with the type set at the specific path.
    ///
    /// # Errors
    /// Because `nest_at_path` doesn't support coalescing or negative indices, an error will
    /// be returned if a path is used that contains either of those. In the future those should
    /// be supported so this functon can just return `Kind` instead of `Result`.
```